packages/contracts/contracts/SimpleLockESS.sol

```diff
@@ -0,0 +1,144 @@
+     *         forfeited.
+     */
+    uint256 public immutable upgradeLockTime;
+
+    /**
+     * @notice Mapping of ChugSplashManager addresses to bundle IDs to claimed upgrades.
+     */
+    mapping(address => mapping(bytes32 => ClaimedUpgrade)) public upgrades;
+
+    /**
+     * @param _registry           Address of the ChugSplashRegistry.
+     * @param _executorBondAmount Executor bond amount in ETH.
+     * @param _upgradeLockTime    Amount of time for an executor to complete an upgrade after
+     *                            claiming it.
+     */
+    constructor(
+        ChugSplashRegistry _registry,
+        uint256 _executorBondAmount,
+        uint256 _upgradeLockTime
+    ) {
+        registry = _registry;
+        executorBondAmount = _executorBondAmount;
+        upgradeLockTime = _upgradeLockTime;
+    }
+
+    /**
+     * @notice Allows an executor to post a bond of `executorBondAmount` to claim the sole right to
+     *         execute actions for an upgrade for a period of `upgradeLockTime`. Only the first
+     *         executor to post a bond gains this right. If the executor fails to complete the
+     *         upgrade within the `upgradeLockTime`, a new executor may post a bond and the original
+     *         bond is forfeited and transferred into the wallet of the project’s
+     *         `ChugSplashManager` contract. Note that this strategy creates a PGA for the
+     *         transaction to claim the upgrade but removes PGAs during the execution process.
+     *
+     * @param _project  Address of the ChugSplashManager for the project being claimed.
+     * @param _bundleId ID of the bundle being claimed.
+     */
+    function claim(address _project, bytes32 _bundleId) external payable {
+        require(executorBondAmount == msg.value, "SimpleLockESS: incorrect executor bond amount");
+        require(
+            registry.managers(ChugSplashManager(_project)) == true,
+            "SimpleLockESS: address is not ChugSplashManager"
+        );
+        (ChugSplashBundleStatus status, ) = ChugSplashManager(_project).bundles(_bundleId);
+        require(status == ChugSplashBundleStatus.APPROVED, "SimpleLockESS: bundle is not active");
+
+        ClaimedUpgrade storage upgrade = upgrades[_project][_bundleId];
+
+        uint256 upgradeDeadline = upgrade.timeClaimed + upgradeLockTime;
+        require(block.timestamp > upgradeDeadline, "SimpleLockESS: upgrade in progress");
+
+        address prevSelectedExecutor = upgrade.selectedExecutor;
+
+        upgrade.timeClaimed = block.timestamp;
+        upgrade.selectedExecutor = msg.sender;
+
+        if (prevSelectedExecutor != address(0)) {
+            // If the previously selected executor failed to complete the upgrade, their bond is
+            // sent to the address of the ChugSplashManager.
+            // TODO: Does it make more sense to send the ETH to the owner of the
+            // ChugSplashManager (i.e. the project owner)?
```

Comment:
@smartcontracts What do you think?

Reply:
Nah sending to the `ChugSplashManager` itself is good enough. Relevant to my comment in #50 where we should just have the `ChugSplashManager` be treated kind of like a piggy bank that holds the funds for everything.

Reply:
I also like the idea of seeing a UI that represents changes in balance of the `ChugSplashManager` with an entry like "Account received a fee because this upgrade wasn't completed on time" :wink: